tests/test_codegen_directive.cpp

```
@@ -2,6 +2,114 @@
+    EXPECT_EQ(CodeGen::deriveRuntimeFnName("filesystem", "list_dir"), "__ry_filesystem_list_dir");
+    EXPECT_EQ(CodeGen::deriveRuntimeFnName("path", "get_extension"), "__ry_path_get_extension");
+}
+
+TEST(NativeFnNaming, ErrorGetter) {
+    EXPECT_EQ(CodeGen::deriveRuntimeFnName("base64", "get_last_error"), "__ry_base64_get_last_error");
+    EXPECT_EQ(CodeGen::deriveRuntimeFnName("io", "get_last_error"), "__ry_io_get_last_error");
+}
+
+// --- NativeFnSignature registry tests ---
+
+TEST(NativeFnSigs, RegistryPopulatedAndKeyed) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)